tutorials/kotlin-springboot-app-engine-java8.md

```diff
@@ -0,0 +1,227 @@
+    a substitution, so you might consider setting the `PROJECT_ID` environment
+    variable in your shell.
+
+2.  Enable billing for your project.
+
+3.  Install the [Google Cloud SDK](https://cloud.google.com/sdk/). Make sure
+    you [initialize](https://cloud.google.com/sdk/docs/initializing) the SDK
+    and set the default project to the new project you created.
+
+    Version 175.0.0 or later of the SDK is required.
+
+3.  Install [JDK 8 or higher](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) if you do not already have it.
+
+4.  Install [Maven](https://maven.apache.org/install.html).
+
+## Creating a new app and running it locally
+
+In this section, you will create a new Spring Boot app and make sure it runs. If
+you already have an app to deploy, you can use it instead.
+
+1. Use [start.spring.io](https://start.spring.io) to generate a Spring Boot
+application.
+
+    * Select **Kotlin** as the language and **Maven** as the build system.
+    * Using the Dependency selector, type **GCP Support** and select it.
+    * In the same Dependency selector, type **Web** and select first result.
+    * Click "Switch to the full version" at the bottom to expand the advanced
+      options. Under **Packaging** select **War** in the dropdown.
+
+    > Alternatively, you can [download](https://github.com/GoogleCloudPlatform/kotlin-samples/tree/master/appengine/springboot/) the sample application.
+
+1. Download the generated project and save it to a local folder.
+
+1. Open the resulting project in your favourite IDE or editor and create a new
+source file `MessageController.kt` in the directory
+`src/main/kotlin/com/example/demo` with the following contents:
+    ```kt
+    package com.example.demo
+
+    import org.springframework.web.bind.annotation.RequestMapping
+    import org.springframework.web.bind.annotation.RestController
+
+    data class Message(val text: String, val priority: String)
+
+    @RestController
+    class MessageController {
+        @RequestMapping("/message")
+        fun message(): Message {
+            return Message("Hello from Google Cloud", "High")
+        }
+    }
+    ```
+
+    The package should match that of your group and artifact name.
+
+1.  Edit the file named `DemoApplication.kt` in
+    `src/main/kotlin/com/example/demo` and replace its contents with the
+    following:
+    ```kt
+    package com.example.demo
+
+    import org.springframework.boot.autoconfigure.SpringBootApplication
+    import org.springframework.boot.runApplication
+    import org.springframework.boot.web.servlet.support.SpringBootServletInitializer
+
+    @SpringBootApplication
+    class DemoApplication : SpringBootServletInitializer()
+
+    fun main(args: Array<String>) {
+        runApplication<DemoApplication>(*args)
+    }
+    ```
+
+1. Run the application from the command line using Maven:
+
+        mvn spring-boot:run
```

Comment:
Is this all you need? No application name or anything?

Reply:
nope!